### PR TITLE
Improve `raft.go` comments

### PR DESCRIPTION
**Pull request by neilalexander**
This PR mostly just tries to comment what is going on inside `raft.go` and does odd little bits of housekeeping/cleanup in places.

Signed-off-by: Neil Twigg <neil@nats.io>

### CODE REVIEW

File: server/raft.go [hunk 1]
```diff
@@ -122,108 +122,105 @@ func (state RaftState) String() string {
...
-	// Random generator, used to generate inboxes for instance
-	prand *rand.Rand
+	hh       hash.Hash64  // Highwayhash, used for snapshots
+	snapfile string       // Snapshot filename
+
+	csz   int             // Cluster size
+	qn    int             // Number of nodes needed to establish quorum
+	peers map[string]*lps // Other peers in the Raft group
+
+	removed map[string]struct{}            //
```
Comment (derekcollison) [inline server/raft.go:145]: Missing comment?

File: server/raft.go [hunk 2]
```diff
@@ -380,7 +376,6 @@ func (s *Server) startRaftNode(accName string, cfg *RaftConfig, labels pprofLabe
 		track:    cfg.Track,
 		csz:      ps.clusterSize,
 		qn:       ps.clusterSize/2 + 1,
-		hash:     hash,
```
Comment (derekcollison) [inline server/raft.go:383]: Why removed? Never accessed?

File: server/raft.go [hunk 3]
```diff
@@ -819,7 +859,7 @@ func (n *raft) PauseApply() error {
 		n.stepdown.push(noLeader)
 	}
 
-	n.debug("Pausing our apply channel")
+	n.debug("Pausing our apply queue")
```
Comment (derekcollison) [inline server/raft.go:862]: These are tough, sometimes easy to see a better log line but if its out there folks could depend on it, so I usually shy away from changing.

File: server/raft.go [hunk 4]
```diff
@@ -837,7 +879,7 @@ func (n *raft) ResumeApply() {
 		return
 	}
 
-	n.debug("Resuming our apply channel")
+	n.debug("Resuming our apply queue")
```
Comment (derekcollison) [inline server/raft.go:882]: Same.

File: server/raft.go (hunk 1 above)
Comment (neilalexander) [reply server/raft.go]: Forgot to fill those in, have done so now.

File: server/raft.go (hunk 2 above)
Comment (neilalexander) [reply server/raft.go:383]: Yep, it was never used.

File: server/raft.go (hunk 3 above)
Comment (neilalexander) [reply server/raft.go]: Good point, I've put them back to what they were.

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**